### PR TITLE
stop caching probed storage data

### PR DESCRIPTION
**Pull request by mwhudson**
A recent branch carefully called Prober.get_storage() whenever a udev
block device event was seen. But Prober.get_storage() cached the results
from probert rendering that useless. Rip the caching and some other
complexity out of the prober class.

### CODE REVIEW

File: subiquitycore/prober.py [hunk 1]
```diff
@@ -15,45 +15,38 @@
...
     def probe_network(self, receiver):
-        if self.opts.machine_config:
+        if self.saved_config is not None:
             observer = StoredDataObserver(
                 self.saved_config['network'], receiver)
         else:
             observer = UdevObserver(receiver)
         return observer, observer.start()
 
     def get_storage(self, probe_types=None):
```
Comment (raharper) [inline subiquitycore/prober.py]: Hrm, I do wish we called this probe_storage() to match probe_network().  But that's not a big deal.

Comment (mwhudson) [reply subiquitycore/prober.py]: One day...

**Comment by mwhudson** [comment]
Tested on real hw and it worked fwiw

**Comment by xnox** [comment]
AHA

**Pull request merged**